src/test/java/com/google/devtools/build/lib/actions/ResourceManagerTest.java

```diff
@@ -167,6 +199,13 @@ public void testOverBudgetRequests() throws Exception {
     assertThat(rm.inUse()).isTrue();
     release(0, 0, bigTests);
     assertThat(rm.inUse()).isFalse();
+
+    // Ditto, for extra resources (even if they don't exist in the available resource set):
+    ImmutableMap<String, Float> bigExtraResources = ImmutableMap.of("gpu", 10.0f, "fancyresource", 10.0f, "nonexisting", 10.0f);
```

Comment:
I think if we try to use non existing resource we should return an error.

Reply:
Or the resource availability can be added first.